java/src/jmri/managers/AbstractProxyManager.java

```diff
@@ -299,29 +299,24 @@ public String createSystemName(@Nonnull String curAddress, @Nonnull String prefi
         return createSystemName(curAddress, prefix, getNamedBeanClass());
     }
 
-    public String getNextValidAddress(@Nonnull String curAddress, @Nonnull String prefix, boolean ignoreInitialExisting, char typeLetter) throws jmri.JmriException {
+    @Nonnull
+    public String getNextValidSystemName(@Nonnull NamedBean currentBean) throws JmriException {
+        int prefixLength = Manager.getSystemPrefixLength(currentBean.getSystemName());
+
+        String prefix = currentBean.getSystemName().substring(0, prefixLength);
+        char typeLetter = currentBean.getSystemName().charAt(prefixLength);
+
         for (Manager<E> m : mgrs) {
-            log.debug("NextValidAddress requested for {}", curAddress);
+            log.debug("getNextValidSystemName requested for {}", currentBean.getSystemName());
             if (prefix.equals(m.getSystemPrefix()) && typeLetter == m.typeLetter()) {
                 try {
-                    switch (typeLetter) { // use #getDefaultManager() instead?
-                        case 'T':
-                            return ((TurnoutManager) m).getNextValidAddress(curAddress, prefix, ignoreInitialExisting);
-                        case 'S':
-                            return ((SensorManager) m).getNextValidAddress(curAddress, prefix, ignoreInitialExisting);
-                        case 'L':
-                            return ((LightManager) m).getNextValidAddress(curAddress, prefix, ignoreInitialExisting);
-                        case 'R':
-                            return ((ReporterManager) m).getNextValidAddress(curAddress, prefix, ignoreInitialExisting);
-                        default:
-                            return null;
-                    }
+                    return ((NameIncrementingManager) m).getNextValidSystemName(currentBean);
                 } catch (jmri.JmriException ex) {
                     throw ex;
                 }
```

Comment:
No need to change, but these three lines don't seem to do anything useful.

Reply:
Thanks for catching, that's an idiom that's been around since Java 1 and isn't needed any more.